### PR TITLE
Remove RESUME trigger handling

### PR DESCRIPTION
**Pull request by ranj063**

### CODE REVIEW

**Review by RanderWang — COMMENTED** [review]
could you explain "full resume" ? Thanks!

**Comment by plbossart** [comment]
> could you explain "full resume" ? Thanks!

Agree with @RanderWang. We should state that the driver does not support the INFO_RESUME capability and resuming needs to be handled by the ALSA core with the .prepare and .trigger_start stages.

**Comment by ranj063** [comment]
> > could you explain "full resume" ? Thanks!
> 
> Agree with @RanderWang. We should state that the driver does not support the INFO_RESUME capability and resuming needs to be handled by the ALSA core with the .prepare and .trigger_start stages.

done. thanks!

File: sound/soc/sof/pcm.c [hunk 1]
```diff
@@ -400,26 +400,6 @@ static int sof_pcm_trigger(struct snd_soc_component *component,
 	case SNDRV_PCM_TRIGGER_PAUSE_RELEASE:
 		stream.hdr.cmd |= SOF_IPC_STREAM_TRIG_RELEASE;
 		break;
-	case SNDRV_PCM_TRIGGER_RESUME:
```
Comment (plbossart) [inline sound/soc/sof/pcm.c:403]: typo in commit message: iremove.

**Review by plbossart — CHANGES_REQUESTED** [review]

**Review by plbossart — APPROVED** [review]

**Review by kv2019i — CHANGES_REQUESTED** [review]
Please add a reference to "ASoC: SOF: pcm: do not add SNDRV_PCM_INFO_RESUME to runtime hw info". This looks really odd otherwise. Why all this code if the feature is not supported.

**Review by RanderWang — DISMISSED** [review]

**Comment by bardliao** [comment]
So driver will return error if it got trigger resume? What is the benefit of this PR?

**Comment by plbossart** [comment]
> So driver will return error if it got trigger resume? What is the benefit of this PR?

we can't have TRIGGER_RESUME because INFO_RESUME is not supported @bardliao. So the ALSA core will instead use .prepare and TRIGGER_START, as for xrun handling. The only difference between resume and xrun is that in the latter case we don't need to reinitialize any hardware, whereas in the resume case the device might have been power-gated or reset.

**Comment by plbossart** [comment]
> Please add a reference to "ASoC: SOF: pcm: do not add SNDRV_PCM_INFO_RESUME to runtime hw info". This looks really odd otherwise. Why all this code if the feature is not supported.

yes, if we had two developers that missed the idea we probably need to expand on the explanations @ranj063. Thanks!

**Comment by ranj063** [comment]
> > Please add a reference to "ASoC: SOF: pcm: do not add SNDRV_PCM_INFO_RESUME to runtime hw info". This looks really odd otherwise. Why all this code if the feature is not supported.
> 
> yes, if we had two developers that missed the idea we probably need to expand on the explanations @ranj063. Thanks!

will do. But given that the change is because of a common flag in hw_info, do you still want 3 patches? I think we can squash them all into one, right?

**Comment by plbossart** [comment]
> will do. But given that the change is because of a common flag in hw_info, do you still want 3 patches? I think we can squash them all into one, right?

Better to have separate patches, one change is for the core and the rest if for Intel stuff.

**Review by plbossart — APPROVED** [review]

**Comment by plbossart** [comment]
let's give @bardliao and @RanderWang the opportunity to review again.

**Review by RanderWang — APPROVED** [review]
LGTM

**Review by bardliao — APPROVED** [review]
Thanks @ranj063

**Pull request merged**